rpc-server/src/utils.rs

```diff
@@ -28,61 +18,87 @@ where
         ),
     };
+    match near_rpc_client.call(block_request_method).await {
+        Ok(block_view) => {
+            // Updating the metric to expose the block height considered as final by the server
+            // this metric can be used to calculate the lag between the server and the network
+            // Prometheus Gauge Metric type do not support u64
+            // https://github.com/tikv/rust-prometheus/issues/470
+            crate::metrics::FINAL_BLOCK_HEIGHT
+                .set(i64::try_from(block_view.header.height).unwrap());
 
-    let block_response = near_rpc_client.call(block_request_method).await?;
-
-    // Updating the metric to expose the block height considered as final by the server
-    // this metric can be used to calculate the lag between the server and the network
-    // Prometheus Gauge Metric type do not support u64
-    // https://github.com/tikv/rust-prometheus/issues/470
-    crate::metrics::FINAL_BLOCK_HEIGHT.set(i64::try_from(block_response.header.height)?);
-    Ok(block_response)
-}
-
-pub async fn get_final_cache_block(
-    near_rpc_client: &near_jsonrpc_client::JsonRpcClient,
-) -> Option<CacheBlock> {
-    match get_final_block(near_rpc_client).await {
-        Ok(block_view) => Some(CacheBlock {
-            block_hash: block_view.header.hash,
-            block_height: block_view.header.height,
-            block_timestamp: block_view.header.timestamp,
-            latest_protocol_version: block_view.header.latest_protocol_version,
-            chunks_included: block_view.header.chunks_included,
-            state_root: block_view.header.prev_state_root,
-        }),
+            Some(CacheBlock {
+                block_hash: block_view.header.hash,
+                block_height: block_view.header.height,
+                block_timestamp: block_view.header.timestamp,
+                latest_protocol_version: block_view.header.latest_protocol_version,
+                chunks_included: block_view.header.chunks_included,
+                state_root: block_view.header.prev_state_root,
+            })
+        }
         Err(_) => None,
     }
 }
 
+#[cfg_attr(feature = "tracing-instrumentation", tracing::instrument(skip_all))]
+async fn handle_streamer_message(
+    streamer_message: near_indexer_primitives::StreamerMessage,
+    blocks_cache: std::sync::Arc<std::sync::RwLock<lru::LruCache<u64, CacheBlock>>>,
+    final_block_height: std::sync::Arc<std::sync::atomic::AtomicU64>,
+) -> anyhow::Result<()> {
+    let block = CacheBlock {
+        block_hash: streamer_message.block.header.hash,
+        block_height: streamer_message.block.header.height,
+        block_timestamp: streamer_message.block.header.timestamp,
+        latest_protocol_version: streamer_message.block.header.latest_protocol_version,
+        chunks_included: streamer_message.block.header.chunks_included,
+        state_root: streamer_message.block.header.prev_state_root,
+    };
+    tracing::debug!("Block {:?}", block);
```

Comment:
This log message is too implicit. And it will be quite noisy. I don't think it makes a lot of sense as we set the metric below which gives us an understanding whether it is working or not. 

```suggestion
```